### DISCUSSION TITLE
Rematch in the grand final.

### DISCUSSION THREAD

**Discussion by DavidAlejandroSaenzMeneses** [Drarig29/brackets-viewer.js#102] [General]
¿The library can generate an additional grand final as a rematch?; in the double elimination, all participants have a second chance. If a participant reaches the grand final without losing and loses only in that match, they should be able to have a second opportunity in that rematch match.
Thanks!

**Comment by DavidAlejandroSaenzMeneses** [comment]
<img width="636" alt="DBracket" src="https://github.com/Drarig29/brackets-viewer.js/assets/61164104/8e5bbbb0-748a-428d-8b22-4f103ccda2f7">

**Comment by Drarig29** [comment]
Yes, you can use `grandFinal: 'double'`.

https://drarig29.github.io/brackets-docs/reference/model/interfaces/StageSettings.html#grandFinal

**Reply by DavidAlejandroSaenzMeneses** [reply]
I have applied that configuration to the stage, but it relates to the loser of the LB semi1. The mentioned rematch would involve the same participants from the grand final.
<img width="209" alt="DB test 2" src="https://github.com/Drarig29/brackets-viewer.js/assets/61164104/df94d819-1219-4cf0-bfd5-87e254c5c30e">

**Reply by Drarig29** [reply]
The consolation final is a separate setting. By default, you only see one match for the grand final, and a second one will appear if required.

**Reply by DavidAlejandroSaenzMeneses** [reply]
I have the configuration (grandFinal: 'double') in the stage, but what appears is the consolation bracket. Do I need to do something additional?
The stage object:
```md
{
      id: 0,
      tournament_id: 0,
      name: "",
      type: "double_elimination",
      number: 1,
      settings: {
        size: totalNumberPartitipants,
        seedOrdering: ordering,
        grandFinal: "double",
        matchesChildCount: 0,
        GrandFinal: 'double',
        GrandFinalType: 'double',//test
      },
    }
```
Thanks.

**Reply by Drarig29** [reply]
You will have 3 final matches if the library thinks it's the right thing to do:
- 2 matches of grand final
- 1 match of consolation final

The 2nd match of grand final **will only be added if required**. For now, you only have 1 match of grand final and 1 match of consolation final.

That's normal.

**Comment by DavidAlejandroSaenzMeneses** [comment]
These are the steps I have taken to implement the double or rematch grand final:
1 - Add ``` grandFinal : 'double'``` to stage settings.
2 - Add a new round (id: 12) related to the last group (group_id: 2).
3 - Add a new match related to the group and round mentioned in point 2 
```md
{
            "id": 31,
            "number": 2,
            "stage_id": 0,
            "group_id": 2,
            "round_id": 12,
            "child_count": 0,
            "status": 0,
            "opponent1": {
                "id": null
            },
            "opponent2": {
                "id": null
            }
        }
```

Are these steps sufficient? Thank you for the guidance, and I apologize for any inconvenience.
Note: I have conducted tests based on the example with-local-storage from this repository.

**Reply by Drarig29** [reply]
Can you send actual code of what you did?

**Comment by DavidAlejandroSaenzMeneses** [comment]
Of course, this is the db.json file with the modifications from the previous comment. I am testing the demo with printing.
[db_modified.json](https://github.com/Drarig29/brackets-viewer.js/files/13920217/db_modified.json)

**Reply by Drarig29** [reply]
@DavidAlejandroSaenzMeneses so do you want to print what's rendered by the viewer?

What I'm trying to explain is that currently, the viewer takes the decision to not show any second grand final match until it's the right time to show it:

https://github.com/Drarig29/brackets-viewer.js/blob/5bdcdaa16b285694252412e6ad337bf630d08738/src/main.ts#L451

So if you have the following in your `db.json` file:

```jsonc
... // all the other matches
{
    "id": 29, // grand final, round 1
    "number": 1,
    "stage_id": 0,
    "group_id": 2,
    "round_id": 10,
    "child_count": 0,
    "status": 0,
    "opponent1": {
        "id": null
    },
    "opponent2": {
        "id": null,
        "position": 1
    }
}
```

You'll have:

![image](https://github.com/Drarig29/brackets-viewer.js/assets/9317502/f7d7ebd7-700c-4708-a65b-656d5e9b4b8a)

---

Now, if you make the following change:
```diff
{
... // all the other matches
{
    "id": 29, // grand final, round 1
    "number": 1,
    "stage_id": 0,
    "group_id": 2,
    "round_id": 10,
    "child_count": 0,
    "status": 0,
    "opponent1": {
-        "id": null
+        "id": 0
    },
    "opponent2": {
        "id": null,
        "position": 1
    }
}
```

You'll get:

![image](https://github.com/Drarig29/brackets-viewer.js/assets/9317502/ac1262c9-396c-458e-a8a4-817ad43d609c)

**Comment by DavidAlejandroSaenzMeneses** [comment]
Thank you for the assistance provided. On this occasion, I need to generate the third-place match or consolation match. To achieve this, I added the configuration 'consolationFinal: true,' but it is not working correctly. Should the mentioned match be added within the same final round?

**Reply by Drarig29** [reply]
You should not manually add any match yourself. Can you show what it looks like with `consolationFinal: true`?

It used to work in your screenshots 2 months ago

**Reply by DavidAlejandroSaenzMeneses** [reply]
these are my settings:
```md
 settings: {
    size: totalNumberPartitipants,
    seedOrdering: ordering,
    grandFinal: "double",
    consolationFinal: true,
    matchesChildCount: 0,
  }
```

**Reply by DavidAlejandroSaenzMeneses** [reply]
With consolationFinal: true in single elimination, it looks like this; I added an extra match in the same final round.
<img width="589" alt="db2" src="https://github.com/Drarig29/brackets-viewer.js/assets/61164104/b73d77e9-8d8d-48cf-a576-8012ad2f1293">

**Reply by DavidAlejandroSaenzMeneses** [reply]
But an additional match is necessary, which would be the third-place match, right?
The third-place match should have its own round, or should it be in the same round as the final?

**Reply by Drarig29** [reply]
No, it should not be in the same round as the final. You should have the consolation final in a separate **group** (second one).

**Comment by DavidAlejandroSaenzMeneses** [comment]
Interestingly, in the double elimination, it seems to work correctly; however, in the single elimination, it generates incorrectly as in the previous image.
<img width="787" alt="db3" src="https://github.com/Drarig29/brackets-viewer.js/assets/61164104/fc37c797-c45e-4670-89c3-965a931a8bc5">
